Homework3/Homework3/ViewController.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ViewController.swift
+//  Homework3
+//
+//  Created by Svetlana Safonova on 28.09.2020.
+//  Copyright © 2020 Svetlana Safonova. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet var nameField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+    }
+    
+    enum SegueIdentifire: String {
+        case modal = "ModalPresentation"
+        case push = "PushPresentition"
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard
+            let segueIdentifireRaw = segue.identifier,
+            let segueIdentifire = SegueIdentifire(rawValue: segueIdentifireRaw)
+        else { return }
+        
+        switch segueIdentifire{
+        case .modal:
+            guard
+                let navigationController = segue.destination as? UITabBarController,
+                let controller = navigationController.tabBarController as? TabBarController
+                else { return }
```

Comment:
```suggestion
            guard
                let navigationController = segue.destination as? UITabBarController,
                let controller = navigationController.tabBarController as? TabBarController
            else { return }
```

Для чего здесь guard?